src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -31,20 +33,27 @@
      */
     public AddCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS,
+                        PREFIX_REMARK , PREFIX_TAG);
 
         if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ADDRESS, PREFIX_PHONE, PREFIX_EMAIL)
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
         }
-
+        Remark remark;
+        if (!arePrefixesPresent(argMultimap, PREFIX_REMARK)
+                || !argMultimap.getPreamble().isEmpty()) {
+            remark = new Remark("");
+        } else {
+            remark = ParserUtil.parseRemark(argMultimap.getValue(PREFIX_REMARK).get());
+        }
```

Comment:
@Chesterwongz this part seems to be violating slap, how about this way instead?
```
Remark remark = ParserUtil.parseRemark(argMultimap.getValue(PREFIX_REMARK).orElse(""));
```

Reply:
the part `remark = new Remark("");` is of a different level of abstraction.  It should only be in ParseUtil